web-app/js/portal/data/DataCollection.js

```diff
@@ -131,6 +141,30 @@ Portal.data.DataCollection = function() {
         return linkStore.getRange(); // Get all records
     };
 
+    constructor.prototype.getDownloadLayerName = function() {
+        var firstWfsLink = this.getWfsLayerLinks()[0];
+        var firstWmsLink = this.getWmsLayerLinks()[0];
+        var link = firstWfsLink || firstWmsLink;
+
+        if (link) {
+            var linkName = link.data.name;
+
+            // If layer has no workspace defined, assume it is in the same workspace as the WMS layer
+            if (this._workspaceFromName(linkName)) {
+                return linkName;
+            }
+            else {
+                return this._workspaceFromName(firstWmsLink.data.name) + ':' + linkName;
+            }
+        }
+    };
+
+    constructor.prototype._workspaceFromName = function(layerName) {
```

Comment:
It could be a util thing. The 'knowledge' of workspaces and layer names is from the OGC WMS spec.